src/__tests__/CompareResults/CompareResultsTable.test.tsx

```diff
@@ -0,0 +1,53 @@
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+
+import CompareResultsTable from '../../components/CompareResults/CompareResultsTable';
+import { addFilter } from '../../reducers/FilterCompareResultsSlice';
+import { render, store } from '../utils/setupTests';
+import { screen } from '../utils/test-utils';
+
+describe('Compare Results Table', () => {
+  it('Should match snapshot', () => {
+      render(<CompareResultsTable mode="light" />);
+      
+      expect(document.body).toMatchSnapshot();
+  });
+
+  it('Should open platform options', async () => {
+    render(<CompareResultsTable mode="light" />);
+    const platformFilter = screen.getByTestId('platform-options-button');
+    const user = userEvent.setup({ delay: null });
+
+    await user.click(platformFilter);
+
+    const platformOptionsList = screen.getByTestId('platform-options');
+    expect(platformOptionsList).toBeInTheDocument();
+  });
+
+  it('Should filter data by platform', async () => {
+    const { rerender } = render(<CompareResultsTable mode="light" />);
+    const rows = screen.getAllByTestId('table-row');
+
+    expect(rows).not.toBeNull();
+
+    // const platformFilter = screen.getByTestId('platform-options-button');
+    // const user = userEvent.setup({ delay: null });
+
+    // await user.click(platformFilter);
+
+    // const checkPlatform = screen.getByTestId('macosx1015-64-shippable-qr-checkbox');
+    // await user.click(checkPlatform);
+    
+    // const applyButton = screen.getByTestId('apply-filter');
+    // await user.click(applyButton);
+
+    store.dispatch(addFilter({ name: 'platform', value: 'macosx1015-64-shippable-qr' }));
+
+    await act( async () => rerender(<CompareResultsTable mode="light" />));
+
+    const filteredRows = screen.getAllByTestId('table-row');
+
+    expect(filteredRows.length).toBeNull();
+  });
+
+});
```

Comment:
I've been trying to figure out why this test is failing and so far I have no ideas. I tried using test data for `compareResults` test, but it does not seem to be working. I created a simple reducer `setCompareResults`, and logging statements to track what is going on.

The reducer is being called and a `console.log` statement in the reducer shows it is being set with the test data. However when I call `store.getState().compareResults` from within the test, it always returns the original state.

If we can't resolve this within the next couple of days, I'd recommend filing a ticket to address this and prioritize it after the work week.

Reply:
I made it fail on purpose to see what data is there. I found a way on how to filter the data in a test and updated the pr.